### PR TITLE
User can post runs

### PR DESCRIPTION
**Pull request by jacobhusband**
https://user-images.githubusercontent.com/24307282/202056052-7d7caff5-857e-4d65-8382-1633b5d72d2e.mov

https://user-images.githubusercontent.com/24307282/202056118-891215ad-c6a9-4e31-abd8-59c39a58aa7e.mov

Screenshot 2022-11-15 at 4 46 52 PM

### CODE REVIEW

File: client/pages/home.jsx [hunk 1]
```diff
@@ -1,11 +1,100 @@
...
+      const sortedImgArray = [];
+
+      for (let j = 0; j < 3; j++) {
+        for (let i = 0; i < 3; i++) {
+          if (imgArray[i].order === j + 1) {
+            sortedImgArray.push(imgArray[i]);
+            break;
+          }
+        }
+      }
```
Comment (thebearingedge) [inline client/pages/home.jsx:47]: We can use the built-in array sort method for this. You have a nice bubble sort there, tho.

```js
const sortedImages = imgArray.sort((a, b) => a.order - b.order);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

File: client/pages/post.jsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        for (const key in newPropsPostInfo) {
+          if (newPropsPostInfo[key] === tempId + 1) {
+            swapId = key;
+          }
+        }
+        newPropsPostInfo[`${id}Order`]++;
+        newPropsPostInfo[swapId]--;
+      }
+    }
+
```
Comment (thebearingedge) [inline client/pages/post.jsx:100]: This feels _very_ complicated. I think you are missing some `data-` attributes on your buttons.

```jsx
<button
  onClick={this.swapImageSrc}
  className='right text-light bg-dark'
  data-direction="right"
  data-current-order={index}>
  {rightChevron}
</button>
```

If you had this direction and index information available, would it make manipulating your photo order simpler?

**Review by thebearingedge — CHANGES_REQUESTED** [review]

File: client/pages/post.jsx (hunk 2 above)
Comment (thebearingedge) [reply client/pages/post.jsx:100]: Given the current index and the direction to move, it should be possible to calculate the new indexes for the swapped images.

I'm willing to bet part of the problem is that our state is not holding an array of image objects, which would be much easier to manipulate.

**Comment by jacobhusband** [comment]
Screenshot 2022-11-16 at 11 56 32 AM

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**